production.py

```diff
@@ -39,6 +49,140 @@ def wkhtml_to_pdf(cls, data, options=None):
             'footer-right': '[page]/[toPage]',
             'footer-spacing': '5',
         }
-        return super(ProductionReport, cls).wkhtml_to_pdf(
+        return super(ReportMixin, cls).wkhtml_to_pdf(
             data, options=options
         )
+
+
+class ProductionReport(ReportMixin):
+    """
+    Production Report
+    """
+    __name__ = 'production.report'
+
+
+class ProductionsReport(ReportMixin):
+    """
+    Productions Report
+    """
+    __name__ = 'report.productions'
+
+    @classmethod
+    def parse(cls, report, records, data, localcontext):
+        Production = Pool().get('production')
+
+        domain = ['AND', [
+            'AND', [
+                'OR', [
+                    ('planned_date', '>=', data['start_date']),
+                ], [
+                    ('effective_date', '>=', data['start_date']),
+                ],
+            ], [
+                'OR', [
+                    ('planned_date', '<=', data['end_date']),
+                ], [
+                    ('effective_date', '<=', data['end_date']),
+                ],
+            ]
+        ], [
+            ('state', 'not in', ['request', 'cancel'])
+        ]]
+
+        productions = Production.search(domain)
+        dates = [
+            data['start_date'] + relativedelta(days=d) for d in range(
+                (data['end_date'] - data['start_date']).days + 1
+            )
+        ]
+
+        # Dict object to hold productions with key as date and valeue
+        # as list of productions on that day
+        productions_by_date = dict.fromkeys(dates, [])
+
+        # Group searched productions by date and for a dictionary out
+        # of it
+        productions_grouped_by_date = dict(
```

Comment:
looks awfully complicated for what it does. Simplify. Especially I dont see why you need `(k, list(v))`. Just use the iterator itself.